Graphics/Slicer/Math/Skeleton/NodeTrees.hs

```diff
@@ -109,18 +111,18 @@ sortNodeTrees = sortBy compareNodeTrees
 
 -- dependent utility functions. used by internal components. not exported.
 
--- | Find a node with an output of the PLine given. Start at the most recent generation, and check backwards.
+-- | Find a node with an output of the PLine given. Check the most recent generation, and if recurse is set, check backwards.
 findINodeByOutput :: [[INode]] -> PLine2 -> Bool -> Maybe ([[INode]],INode)
 findINodeByOutput iNodeSets plineOut recurse
   | null iNodeSets = Nothing
   | otherwise = case nodesMatching of
                   [] -> if recurse
                         then case iNodeSets of
                                [] -> Nothing
-                               [_] -> Nothing
+                               [(INode _ _ _ _:_)] -> Nothing
                                (_:_) -> findINodeByOutput (init iNodeSets) plineOut recurse
```

Comment:
Overlapping patterns.  I remarked on a similar issue above, but it appears to be a common mistake in this code base.  Here is an example of a line outside the change set of this branch.  Maybe you want to fix that, too?  It certainly took me a moment to figure out how to read this (then again I have extremely little context).

```haskell
                               [] -> Nothing
                               [(_:_)] -> Nothing
                               [[]] -> findINodeByOutput (init iNodeSets) plineOut recurse
                               (_:_:_) -> findINodeByOutput (init iNodeSets) plineOut recurse
```

Is this still what you intended?  If not I found a bug.  :)  (or introduce one...)

Reply:
good eye. i'll take a look at this. :)

Reply:
ah, no, this is correct, but strangely formatted, due to stan being annoying. it will warn you about using ```[_]``` switching to ```[(_:_)]```.